packages/core/src/tw/model/state.ts

```diff
@@ -40,7 +40,12 @@ export class ModelState implements IModelState {
 
     protected handleDidUpdateStateEvent = (event: IDidUpdateStateEvent) => {
         const wrappedDepth = this.findWrappedDepth(this.stateService.getTokens(), event.afterFrom, event.afterTo);
-        const node = this.findNodeContainingRange(event.beforeFrom, event.beforeTo, wrappedDepth);
+        let node
+        if (identifyTokenType(this.stateService.getTokens()[event.beforeFrom]) === 'OpenToken') {
+            node = this.docNode.resolvePosition(event.beforeFrom).getNode()
```

Comment:
This doesn't seem correct. What if the range covers more than one node? Then the common updated node would be the parent.

Although, I do plan to refactor how state transformations are applied. I think maintaining a flat token array state is redundant. The current implementation is also extremely inefficient when adding / removing blocks, since it causes the whole document to be updated. This piece of logic will likely be replaced as part of the refactor, as it is core to the inefficiency.